qiime/core/type/signature.py

```diff
@@ -94,10 +110,30 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    @property
+    def descriptions(self):
+        params = {**self.inputs, **self.parameters, **self.outputs}
```

Comment:
I'm not sure on the ordering. Would the only issue be if someone named an output the same as the others? Because being a python function, the inputs and parameters should all be restricted to needing different names, but if our function registration allows outputs to be named the same as an input I guess it could cause issues. Otherwise all the descriptions are just being grabbed by the name of the input/parameter/output, so I don't know for sure if the ordering would affect it, as it's not just being iterated over? I don't know though.